src/macros.rs

```diff
@@ -0,0 +1,207 @@
+/// Utility implementations
+
+/// Implements common functions for a Socket wrapper.
+macro_rules! impl_wrapper {
+    ($type: ty, $target: ty, $field: ident) => {
+        impl $crate::socket::AsZmqSocket for $type {
+            #[inline]
+            fn get_socket(&self) -> &::zmq::Socket {
+                &self.$field.get_socket()
+            }
+        }
+        impl $type {
+            #[doc(hidden)]
+            #[inline]
+            pub fn into_inner(self) -> $target {
+                self.$field
+            }
+        }
+    };
+}
+
+/// Implements `Sink<Multipart, Error=TmqError>` for a Socket wrapping an inner `Sender`.
+macro_rules! impl_wrapper_sink {
+    ($type: ty, $field: ident) => {
+        impl<T: ::std::convert::Into<$crate::Multipart>> ::futures::Sink<T> for $type {
+            type Error = $crate::TmqError;
+
+            #[inline]
```

Comment:
Yes, in general if you write thin oneline wrappers it might be a good idea to use `#[inline]`. Actually I might even use `#[inline(always)]` on these onelines, because recently I got bit by a perf. regression because a short function marked with `#[inline]` wasn't being inlined.

Reply:
Of course it also has some costs, notably increasing compilation time. But we have a lot of layers of indirections here and without LTO it could cause issues if everything was a function call. So I would keep it.